proposals/0002-gep.md

```diff
@@ -0,0 +1,58 @@
+---
+MEP: 0002
+Title: Unifed API
+Discussion: Link
+Implementation: Link
+---
+
+# TITLE: Unified API
+
+## Abstract
+
+This GEP discusses the unified API that handles the interaction with LLM providers.
+
+## Motivation
+
+[TBU, what problem we are trying to solve, focus on users, use cases, and all important context]
+
+The Unified API holds all the logic needed to create a request to a provider. This solution allows the clients to be easily
+envoked by a provider pool.
+
+
+### Requirements
+
+R1: User should be able to specify specific models or select auto mode (selection handled by Glide)
+R2: Should be a Hertz Client
+R3: Should take a user request as input
+R4: Handles all provider specific logic
+R5: Easily maintained
+
+
+## Design
+
+[TBU, all information about the proposed solution, possibly evaluations against the requirements]
+
+- A Hertz client that handles an incoming request and returns a reponse from that provider.
+- Provider specific clients
+
+### Routes
+
+### Request Schemas
+
+### Response Schemas
+
+## Example
+
+User request calls for a fallback strategy of  OpenAI backed up by Anthropic. OpenAI is available so the OpenAI client is invoked and OpenAI reuqest is built from data in the user request.
+
+## Reference
+
+LangchainGo: https://github.com/tmc/langchaingo/tree/main
+
+## Alternatives Considered
+
+[TBU, what other solutions were considered and why they were rejected]
+
+## Future Work
+
+[TBU, what we put outside of the equation now, but may want to consider in the future]
```

Comment:
As a future work, we can mention streaming responses. Don't think we need to touch them right now. Too early to think about that.